### DISCUSSION TITLE
No PV Data in Energy Distribution

### DISCUSSION THREAD

**Discussion by Captain0Kangaroo** [StealthChesnut/HA-FoxESS-Modbus#6] [General]
Seems i'm missing the PV data in the Energy Distribution in HA.

Its logging in Solar Production but not in the energy side of things?
<img width="1462" alt="Screenshot 2022-05-05 at 08 52 15" src="https://user-images.githubusercontent.com/96144685/166882733-aa12595f-2fd6-4359-8b08-5f8a5bafe8ba.png">

<img width="1163" alt="Screenshot 2022-05-05 at 09 04 25" src="https://user-images.githubusercontent.com/96144685/166884155-8d09150e-8ce5-4134-a74d-8519c3130d2e.png">

**Comment by Daibutt** [comment]
Look at the scales on the graphs. Possibly too small to show on the top one? 
Maybe see what it's like when you're generating PV.

**Reply by Captain0Kangaroo** [reply]
It's my fault i think. I had the wrong sensor in the Energy Config. Not sure how but it was pv1_sum and not pv1_daily. Same for pv2.

Feel free to delete this post:

**Comment by Captain0Kangaroo** [comment]
Strange... Its been a few hours and still now PV data in the energy consumption however i'm seeing it still under Solar Production.

Any ideas? I thought it was due to the wrong sensor but maybe not.

**Reply by williamjeccles** [reply]
@Captain0Kangaroo Double-check you've replaced the whole modbus.yaml and applied the new configuration file. There are suttle differences. Once you done that check that you see data in PV1_Power, PV2_Power and PV_Power_now

**Reply by Captain0Kangaroo** [reply]
Replaced the code just to triple check and still nothing. The odd thing is that it was working yesterday.

**Reply by williamjeccles** [reply]
@Captain0Kangaroo when you say nothing, are all three values empty for PV1_Power, PV2_Power and PV_Power_now?

**Reply by Captain0Kangaroo** [reply]
As you can see i've got data for production which is gathering data from PV1_Power, PV2_Power and PV_Power_now i'm assuming. Its just not logging it in the energy consumption.
<img width="500" alt="Screenshot 2022-05-05 at 16 02 14" src="https://user-images.githubusercontent.com/96144685/166953032-241bbd7e-c781-48f9-bf43-6f49a7a08817.png">

<img width="1457" alt="Screenshot 2022-05-05 at 16 03 26" src="https://user-images.githubusercontent.com/96144685/166953276-b3f8f416-0f2c-41a8-80ea-48e7c9b3543b.png">

**Reply by StealthChesnut** [reply]
The data in your graph at the bottom is showing values like 0.003 kWh - 3 Watt-hours? I think you may have something odd happening where your PV data is coming in, but then being mislabelled. So 1kW reported from your inverter is becoming 1W in the home assistant pv1_sum entity somewhere along the way.

This isn't easy to fix, but if we can your current config may then work fine. Could you take a look at this forum post and see if you are able to find any discrepancies in your DB? https://community.home-assistant.io/t/how-to-delete-old-energy-data/338305

**Reply by Captain0Kangaroo** [reply]
Looks easy - NOT 🤣

This seems to have happened after the last release which I updated last night as yesterdays data is all fine.

Think I'll just kill this HA test (currently in a VM) and try again.

Just concerned when implementing into my live HA 🤔

**Reply by StealthChesnut** [reply]
Yes - changing entities which feed the Energy dashboard isn't easy. If you're able to recreate a VM, pull the latest and try again it would be good to see if the current version works for you.

**Reply by Captain0Kangaroo** [reply]
So did a complete fresh HA install last night and left it running until just now so it's had time to populate data in the sensors (battery charge, discharge, solar generation on both arrays)

pv1_daily is there but pv2_daily doesn't show up.

bat_discharge is showing but bat_charge is not showing up.


Ive literally just copied the config line for line with no mods and the same for the modbus obviously using my IP address.

?????

**Reply by StealthChesnut** [reply]
ok, so the pv1_daily and pv2_daily are Home Assistant Utility Meters - https://www.home-assistant.io/integrations/utility_meter/
As it says there, the first cycle (which is what you're on now - day 0 to day 1) are incomplete. It's a bit frustrating but you may need to wait until tomorrow before we can see if it works properly. It's just the way HA works unfortunately.

If bat_discharge and bat_charge are also the daily meters then the same applies. Until you get some data into them it appears they might not appear as expected?

Are you getting what looks like expected data in these sensors?
pv1_power
pv2_power
battery_charge
batttery_discharge

**Reply by Captain0Kangaroo** [reply]
Lots of data from the sensors. Just can't get them into the energy config.
<img width="324" alt="Screenshot 2022-05-06 at 22 17 57" src="https://user-images.githubusercontent.com/96144685/167217851-9a1e24be-f321-451b-876f-92c962fb3981.png">

**Reply by Captain0Kangaroo** [reply]
Still no joy. Its been running now for 36hours with data passing through the system but still can't get all the sensors to come up for the consumption data.

Anyone got any other ideas?

**Comment by Captain0Kangaroo** [comment]
Any chance this might sort itself out on the strike of midnight?

**Comment by Captain0Kangaroo** [comment]
Yay finally i've got it all working. 

I had to rename all the sensors and put '_power' at the end of them so pv1_daily_power etc...

I think maybe the original sensor names were messed up in the HA database and wasn't populating in the Energy Config.
